core/chain.py

```
@@ -1,22 +1,139 @@
         self.state = State()
         self._lock = threading.RLock()
-        self._create_genesis_block()
+        self._chain_file = chain_file or DEFAULT_CHAIN_FILE
+        self._load_from_file()
+
+    def _load_from_file(self):
+        if not os.path.exists(self._chain_file):
+            self._create_genesis_block()
+            return
+
+        temp_file = None
```

Ruff (0.15.1)

[error] Local variable `temp_file` is assigned to but never used

Remove assignment to unused variable `temp_file`

(F841)